### PR TITLE
Feature/ch277/conditional step over

### PR DESCRIPTION
**Pull request by adityagoel4512**
### Clubhouse Ticket
[ch277](https://app.clubhouse.io/manim-dsl/story/277/conditional-step-over)

### Description

Code tracking conditions and annotation style introduced.

### Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe tests added as well.

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by adityagoel4512** [comment]
I split the tasks so the annotations to do with speed will be separate

**Comment by mayanksharma3** [comment]
I'm quite baffled as to how, but somehow this commit https://github.com/ManimDSL/ManimDSLCompiler/commit/810c8f478993df1042c0d52a53509ee6e66a211c has ended up on this branch?

File: src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt [hunk 1]
```diff
@@ -210,9 +210,16 @@ class VirtualMachine(
...
-                EmptyValue
+                val condition = executeExpression(statement.condition)
+                if (condition is BoolValue) {
+                    previousStepIntoState = stepInto
+                    if (condition.value) {
+                        stepInto = statement.isStepInto
+                    }
+                    EmptyValue
+                } else {
+                    condition
```
Comment (mayanksharma3) [inline src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt:221]: Why do we return condition if it's not a bool value? Is it because of RuntimeErrors?

Comment (adityagoel4512) [reply src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt:221]: yeah it wasn't implemented before

**Review by mayanksharma3 — APPROVED** [review]

**Comment by mayanksharma3** [comment]
Looks good, I'll add the annotations to the syntax highlighting for the web UI

**Pull request merged**